### ISSUE TITLE
Display events before sliding

### ISSUE THREAD

**Issue by radious15** [twinssbc/Ionic2-Calendar#469]
Hello!

Would it be possible to display the events of the next/previous week/day **before** actually starting the slide?

What we currently have (please don't mind the CSS it's just a test calendar): 

![slide_current](https://user-images.githubusercontent.com/44028371/67617786-11d21f00-f7e7-11e9-85b8-ed6370cf72e0.gif)

What I would love to have (Google calendar):

![slide_googlecalendar-expected](https://user-images.githubusercontent.com/44028371/67617584-c585df80-f7e4-11e9-96d8-cf4b110f1a8b.gif)

Thanks for your help!

Best regards,
Radious15

**Comment by twinssbc** [comment]
@radious15 Yes, it's possible. First when you fetch the events, you need to fetch 3 months, instead of 1 month. Second you could use the several weekview inactive templates to customize the inactive views, because by default, it doesn't display any event. More explanation about the templates can be found in README.

**Comment by radious15** [comment]
Hi @twinssbc , thanks for your quick response!

I tried and found few issues (I think):

1. [PreserveScrollPosition] has a weird behavior: when I set this option to true, if I scroll vertically it refreshes my event and it doesn't scroll properly. Also, swiping to next or previous doesn't automatically conserve my scrolling position, sometimes it goes to 0 then scroll, sometimes it starts from 18...

2. Adding InactiveWeekViewTemplate doesn't change anything. How can I display the events the exact same way they are normally displayed? I don't think I got that part.

My code is the following (light test version):

```
<ng-template #templateWeekViewInactive let-tm="tm" let-hourParts="hourParts" let-eventTemplate="eventTemplate">
    <div [ngClass]="{'calendar-event-wrap': tm.events}" *ngIf="tm.events">
      <div *ngFor="let displayEvent of tm.events" class="calendar-event"
        [ngStyle]="{top: (100*displayEvent.startOffset/hourParts)+'px',
        left: 100/displayEvent.overlapNumber*displayEvent.position+'%',
         width: 100/displayEvent.overlapNumber+'%', 
         height: 37*(displayEvent.endIndex -displayEvent.startIndex - (displayEvent.endOffset + displayEvent.startOffset)/hourParts)+'px'}">
        <ng-template [ngTemplateOutlet]="eventTemplate" [ngTemplateOutletContext]="{displayEvent:displayEvent}">
        </ng-template>
      </div>
    </div>
  </ng-template>
```

```
  <calendar [scrollToHour]="calendar.scrollToHour" [eventSource]="eventList" [calendarMode]="calendar.mode"
    [currentDate]="calendar.currentDate" [formatHourColumn]="calendar.formatHourColumn" [startHour]="calendar.startHour"
    [endHour]="calendar.endHour" [step]="calendar.step" [startingDayWeek]="calendar.startingDayWeek"
    preserveScrollPosition=true [weekviewInactiveNormalEventSectionTemplate]="templateWeekViewInactive">
  </calendar>
</ion-content>
```

```
 calendar = {
    mode: 'week',
    currentDate: new Date(),
    startHour: "00",
    scrollToHour: "9",
    endHour: "24",
    step: "30",
    startingDayWeek: "1",
    formatHourColumn: "HH:mm"
  };

```
![issue_scroll_and_slide](https://user-images.githubusercontent.com/44028371/67633181-01847780-f8ad-11e9-94eb-019be21f96c6.gif)

Thanks a lot for your help, I really appreciate! 
Radious

**Comment by twinssbc** [comment]
@radious15 
Regarding your first issue, the abnormal behavior when scrolling vertically, I'm still trying to reproduce.
For the second issue, scrollbar doesn't preserve position, this is because I'm using a looped the Ionic slides control. It's internal implementation has some cached page on the first and last view. I can't control the behavior of these cached pages. That's why when you scroll twice , reach to the third slides, when it goes to the first slide, there will be a sudden change. Unfortunately, it is never fixed by Ionic team.
For the third issue, the events not displayed in the inactive view. Sorry for misleading you. The inactive view is meant to make the inactive view to have the exact layout if you make some customization of the active view. But I didn't consider the requirement that the events need also be displayed in the inactive view. I will add an option to display inactive view events.

**Comment by radious15** [comment]
@twinssbc thank you very much! I am going to deep test the scrolling issue and let you know if I find any workaround. Thanks for the event, it would be perfect to see them when scrolling! Have a nice day :)
